apps/web/pages/index.vue

```diff
@@ -1,48 +1,92 @@
-        <section class="mb-10 overflow-hidden">
-          <p data-testid="recommended-products" class="mb-4 typography-text-lg text-center md:text-left">
-            {{ item.headline }}
-          </p>
-          <ProductRecommendedProducts cache-key="homepage" :category-id="item.categoryId" />
-        </section>
+  <div>
+    <Editor
+      v-if="isEditing && currentBlockIndex !== null"
+      :index="currentBlockIndex"
+      :block="currentBlock"
+      @update="updateBlock"
+    />
+    <div v-else class="content">
+      <template v-for="(block, index) in data.blocks" :key="index">
+        <div
+          :class="[
+            'relative max-w-screen-3xl mx-auto md:px-6 lg:px-10 mb-10 group',
+            { 'border-[3px] border-[#538AEA]': disableActions && isClicked && isTablet && clickedBlockIndex === index },
+            { 'hover:border-[3px] hover:border-[#538AEA]': disableActions && !isTablet },
+          ]"
+          @click="tabletEdit(index)"
+        >
+          <UiBlockActions v-if="disableActions && blockHasData(block)" :index="index" @edit="handleEdit" />
+          <component :is="getComponent(block.name)" v-bind="block.options" />
+        </div>
       </template>
-
-      <NuxtLazyHydrate when-visible>
-        <NewsletterSubscribe v-if="showNewsletter" />
-      </NuxtLazyHydrate>
     </div>
   </div>
 </template>
 
 <script lang="ts" setup>
-const { isEditing } = useEditor();
+import { Block } from '~/composables/useHomepage/types';
+
+const { isEditing, disableActions } = useEditor();
+const viewport = useViewport();
+
+const { data, fetchPageTemplate } = useHomepage();
 const { fetchCategoryTemplate } = useCategoryTemplate();
-const { data: homepage, fetchPageTemplate } = useHomepage();
-const { showNewsletter } = useNewsletter();
+
+const currentBlock = ref<Block | null>(null);
+const currentBlockIndex = ref<number | null>(null);
+const isClicked = ref(false);
+const clickedBlockIndex = ref<number | null>(null);
+
+const isTablet = computed(() => viewport.isLessThan('lg') && viewport.isGreaterThan('sm'));
+const isEmptyBlock = (block: Block): boolean => {
+  const options = block?.options;
+  return !options || (typeof options === 'object' && Object.keys(options).length === 0);
+};
+const blockHasData = (block: Block): boolean => {
+  return !isEmptyBlock(block);
+};
+const tabletEdit = (index: number) => {
+  if (isTablet.value) {
+    isClicked.value = !isClicked.value;
+    clickedBlockIndex.value = isClicked.value ? index : null;
+  }
+};
+const handleEdit = (index: number) => {
+  if (data.value.blocks && data.value.blocks.length > index) {
+    currentBlockIndex.value = index;
+    currentBlock.value = data.value.blocks[index];
+    isEditing.value = true;
+  }
+};
+
+const updateBlock = (index: number, updatedBlock: Block) => {
+  if (data.value.blocks && index !== null && index < data.value.blocks.length) {
+    data.value.blocks[index] = updatedBlock;
+  }
+};
```

Comment:
**note**: For now this is fine, but at some point in the future we should move these actions to a composable.

Reply:
Will be addressed separately.